src/wallet/wallet.cpp

```diff
@@ -1418,6 +1429,13 @@ bool CWallet::IsHDEnabled() const
     return !hdChain.seed_id.IsNull();
 }
 
+bool CWallet::IsBlank()
+{
+    // A Blank wallet either has the flag set or it has a version number requiring HD but no HD seed is set
+    LOCK(cs_wallet);
+    return IsWalletFlagSet(WALLET_FLAG_BLANK_WALLET) || (CanSupportFeature(FEATURE_HD) && !IsHDEnabled());
```

Comment:
In commit "[wallet] Support creating a blank wallet" (8061ea4611696b22d5d5dcdd73c1e053ffecc09d)

I think there is no point to checking WALLET_FLAG_BLANK_WALLET here. It's redundant and just complicates the logic.

Also, I don't think this function should be called IsBlank, because this creates two slightly different definitions of blank, one where a wallet stops being blank when a watch-only address is added (WALLET_FLAG_BLANK_WALLET), and one which remains blank (IsBlank). I think a more accurate name for this function would be CannotGenerateKeys, but even better would be:

```c++
bool CWallet::CanGenerateAddresses()
{
    // Wallet can generate addresses if it has an HD seed, or is a pre-hd wallet.
    return IsHDEnabled() || !CanSupportFeature(FEATURE_HD);
}
```

which could simplify all the places where this is called, as well as the corresponding GUI function.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15226#discussion_r255249959

After seeing #14075, I'd tweak my suggestion above and call this `CanGenerateKeys` or `CanTopUpKeyPool` instead of `CanGenerateAddresses`. After #14075 it looks like it will be possible to generate addresses with imported keys even if the wallet can't generate new keys.